listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -493,6 +503,76 @@ def unhide_user_timeline_event(user_name):
     return jsonify({"status": "ok"})
 
 
+@user_timeline_event_api_bp.route('/user/<user_name>/timeline-event/create/recommend-personal', methods=['POST', 'OPTIONS'])
+@crossdomain
+@ratelimit()
+def create_personal_recommendation_event(user_name):
+    '''
+    Make the user recommend a recording to their followers.
+    The request should post the following data about the recording being
+    recommended, and also the list of followers getting recommended:
+
+    .. code-block:: json
+    {
+        "metadata": {
+            "artist_name": "<The name of the artist, required>",
+            "track_name": "<The name of the track, required>",
+            "recording_msid": "<The MessyBrainz ID of the recording, required>",
+            "release_name": "<The name of the release, optional>",
+            "recording_mbid": "<The MusicBrainz ID of the recording, optional>",
+            "followers": [<Row IDs (integer) of the follower, required>]
+            "blurb_content": "<String containing personalized recommendation>"
+        }
+    }
+
+    :statuscode 200: Successful query, recording has been recommended!
+    :statuscode 400: Bad request, check ``response['error']`` for more
+    details.
+    :statuscode 401: Unauthorized, you do not have permissions to recommend
+    personal recordings on the behalf of this user
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+
+    '''
+    user = validate_auth_header()
+    events = []
+
+    try:
+        data = ujson.loads(request.get_data())
+    except ValueError as e:
+        raise APIBadRequest(f"Invalid JSON: {str(e)}")
+
+    metadata = data['metadata']
+
+    try:
+        metadata_db = {
+            "artist_name": metadata["artist_name"],
+            "track_name": metadata["track_name"],
+            "recording_msid": metadata["recording_msid"],
+            "release_name": metadata["release_name"],
+            "recording_mbid": metadata["recording_mbid"],
+            "recommendee_id": 0,
```

Comment:
I am unsure about this one. If we assign recommendee_id later on then it won't be validated. But if we do it for each user then the remaining fields are validated multiple times. That is a bit wasteful but unless we know its a noticeable slowdown, I suggest that we validate recommendee_id as well.

To be clear, validate each metadata first and throw a 400 if any validation error happens. After all are validated, then start inserting in database.